soc/arm/ti_simplelink/cc13x2_cc26x2/power.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2019 Linaro Limited.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <init.h>
+#include <power/power.h>
+
+#include <ti/drivers/Power.h>
+#include <ti/drivers/power/PowerCC26X2.h>
+
+#include <ti/drivers/dpl/ClockP.h>
+
+#include <ti/devices/cc13x2_cc26x2/driverlib/cpu.h>
+#include <ti/devices/cc13x2_cc26x2/driverlib/vims.h>
+#include <ti/devices/cc13x2_cc26x2/driverlib/sys_ctrl.h>
+
+#include <logging/log.h>
+#define LOG_LEVEL CONFIG_SOC_LOG_LEVEL
+LOG_MODULE_REGISTER(soc);
+
+/* Configuring TI Power module to not use its policy function (we use Zephyr's
+ * instead), and disable oscillator calibration functionality for now.
+ */
+const PowerCC26X2_Config PowerCC26X2_config = {
+	.policyInitFxn      = NULL,
+	.policyFxn          = NULL,
+	.calibrateFxn       = NULL,
+	.enablePolicy       = false,
+	.calibrateRCOSC_LF  = false,
+	.calibrateRCOSC_HF  = false
+};
+
+extern PowerCC26X2_ModuleState PowerCC26X2_module;
+
+/*
+ * Power state mapping:
+ * SYS_POWER_STATE_SLEEP_1: Idle
+ * SYS_POWER_STATE_SLEEP_2: Standby
+ */
+
+/* Invoke Low Power/System Off specific Tasks */
+void sys_set_power_state(enum power_states state)
+{
+#ifdef CONFIG_SYS_POWER_SLEEP_STATES
+#ifdef CONFIG_HAS_SYS_POWER_STATE_SLEEP_1
+	u32_t modeVIMS;
+	u32_t constraints;
+#endif
+#endif
+
+	LOG_DBG("SoC entering power state %d", state);
+
+	/* Switch to using PRIMASK instead of BASEPRI register, since
+	 * we are only able to wake up from standby while using PRIMASK.
+	 */
+	/* Set PRIMASK */
+	CPUcpsid();
+	/* Set BASEPRI to 0 */
+	irq_unlock(0);
+
+	switch (state) {
+#ifdef CONFIG_SYS_POWER_SLEEP_STATES
+#ifdef CONFIG_HAS_SYS_POWER_STATE_SLEEP_1
```

Comment:
when would `CONFIG_HAS_SYS_POWER_STATE_SLEEP_1` not be defined?

Reply:
Same as above